File: android/src/main/kotlin/com/getcapacitor/plugin/ar/ARPlugin.kt
package com.getcapacitor.plugin.ar

import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.PluginMethod
import com.getcapacitor.PluginCall
import com.getcapacitor.JSObject
import com.getcapacitor.Plugin

@CapacitorPlugin(name = "AR")
class ARPlugin : Plugin() {
    private var implementation: AR = AR(bridge)

    override fun load() {
        implementation = AR(bridge)
    }

    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")
        val ret = JSObject()
        ret.put("value", implementation.echo(value!!))
        call.resolve(ret)
    }

    @PluginMethod
    fun start(call: PluginCall) {
        val ret = JSObject()
        ret.put("started", implementation.start(call))
        call.resolve(ret)
    }
}

File: ios/Plugin/ARPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ARPlugin)
public class ARPlugin: CAPPlugin {
    private var implementation: AR? = nil
    
    override public func load() {
        self.implementation = AR(bridge: self.bridge!)
    }

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation!.echo(value)
        ])
    }
    
    @objc func start(_ call: CAPPluginCall) {
        call.resolve([
            "started": implementation!.start(call)
        ])
    }
}
